### PR TITLE
Feature: Added navigation history flyout when right clicking the back/forward buttons

### LINKED ISSUE #4098: Feature: Show menu when right clicking back/forward buttons to show history
**Is your feature request related to a problem? Please describe.**
It's can be annoying clicking back (or pressing Alt+left arrow) countless times to go back to a folder I was using previously, it would be nice to have a convenient way to view the history for a tab.

**Describe the solution you'd like**
I think right clicking the back or forward buttons should show a popup of the tabs history.

**Describe alternatives you've considered**
Navigate manually. 

**Additional context**
Web browsers have this feature, this is how it is in Edge.
image

### PR DESCRIPTION
**Pull request by d2dyno1**
### Resolved / Related Issues
- Closes #4098

### Changes
This PR introduces highly requested forward and backward navigation history feature. You can access the list by right-clicking either of the toolbar buttons to view a list of recently navigated folders. Selecting any entry will bring you to that folder

### To Do
- [x] Code clean up (sort and move around classes/methods)
- [ ] ~~Implement button holding for touch devices~~
- [x] Add icon for HomePage

### Screenshots
image

### CODE REVIEW

**Comment by yaira2** [comment]
Is `TeachingTip` preferred over `MenuFlyout`?

**Comment by d2dyno1** [comment]
> Is `TeachingTip` preferred over `MenuFlyout`?

It's better because it allows a ListView to be utilized with ItemsSource binding, whereas a MenuFlyout doesn't easily bind to an ObservableCollection of items. Moreover, when button holding is added, MenuFlyout may not work as expected.

I personally think a TeachingTip looks better since it always opens in the same place MenuFlyout on the other hand would open at the cursor's position

**Comment by yaira2** [comment]
I'm not convinced it's the best UX to use here, but it's alright for now.

**Comment by 0x5bfa** [comment]
> it allows a ListView to be utilized with ItemsSource binding

You can utilize MenuFlyout.MenuItemsSource binding.

**Comment by yaira2** [comment]
Preferably, we'd use the actual thumbnails.

**Comment by yaira2** [comment]
@d2dyno1 I found an issue where Local Disk doesn't display the label.
image

**Comment by 0x5bfa** [comment]
> a MenuFlyout doesn't easily bind to an ObservableCollection of items.

Too bad that they didn't make any progress over a half decade `https://github.com/microsoft/microsoft-ui-xaml/issues/1087` but IMO we can create a attached property 'ItemsSource' and put it via MenuFlyout.Items.Add().

> MenuFlyout on the other hand would open at the cursor's position

If it's set to Button.Flyout it should be at the exact correct position.

**Comment by yaira2** [comment]
We can copy what we do for the breadcrumb flyouts

**Review by yaira2 — APPROVED** [review]
Behavior looks good. Can you resolve the merge conflicts?

**Review by 0x5bfa — APPROVED** [review]
Looks good to me, code wise.

File: src/Files.App/UserControls/AddressToolbar.xaml [hunk 1]
```diff
@@ -237,7 +238,7 @@
 
 			<!--  Page Navigation Actions  -->
 			<StackPanel
-				Grid.Row="1"
+				Grid.Column="0"
```
Comment (yaira2) [inline src/Files.App/UserControls/AddressToolbar.xaml:241]: Was this changed on purpose?

Comment (d2dyno1) [reply src/Files.App/UserControls/AddressToolbar.xaml:241]: Yes, Grid.Row showed a warning (there are no rows defined)

Comment (yaira2) [reply src/Files.App/UserControls/AddressToolbar.xaml:241]: Thanks

**Pull request merged**